src/main/scala/com/sky/kafka/topicloader/config/config.scala

```diff
@@ -16,7 +17,8 @@ package object config {
   implicit class EitherOps[A](e: Either[Throwable, A]) {
     def validate(path: String): ValidationResult[A] = e.leftMap {
       case ce: ConfigException => ce
-      case e: Throwable        => new ConfigException.BadValue(path, e.getMessage)
+      case e if NonFatal(e)    => new ConfigException.BadValue(path, e.getMessage)
```

Comment:
I think you can shorten this to just be
```suggestion
      case NonFatal(e)  => new ConfigException.BadValue(path, e.getMessage)
```